laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -249,109 +249,57 @@ public function getDay($day = null, $room = null, $leadSurgeon = null, $rotation
             return view('nonpermit');
         }
         // get the id from the form
         $id = $_REQUEST['schedule_id'];
-
+        $choiceTypes = array("First", "Second", "Third");
 		// id is stored as id1_id2_id3, need to split it to get the individual ids
 		$split = explode("_", $id);
-
-		//start with the first choice
-		$choice = 1;
-		// get the first choices data
-        $schedule_data1 = ScheduleData::where('id', $split[0])->get();
-        $input[0] = array(
-            'id'=>$split[0],
-            'schedule' => ScheduleData::where('id', $split[0])->get(),
-            'choice'=>$choice,
-            'milestones'=>Milestone::where('id', $_REQUEST['milestones1'])->get(),
-            'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest1'])->get(),
-            'objectives'=>$_REQUEST['objectives1']
-        );
-
-    	// get the second choices data
-    	$choice++;
-    	// check if the second choice exists
-        if ($split[1] != 0){
-        	$schedule_data2 = ScheduleData::where('id', $split[1])->get();
-            $input[1] = array(
-                'id'=>$split[1],
-                'schedule'=>ScheduleData::where('id', $split[1])->get(),
-                'choice'=>$choice,
-                'milestones'=>Milestone::where('id', $_REQUEST['milestones2'])->get(),
-                'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest2'])->get(),
-                'objectives'=>$_REQUEST['objectives2']
-            );
-        } else {
-            $input[1] = NULL;
-        }
-
-
-    	// get the third choices data
-        $choice++;
-    	// check if the third choice exists
-        if ($split[2] != 0){
-    		$schedule_data3 = ScheduleData::where('id', $split[2])->get();
-            $input[2] = array(
-                'id'=>$split[2],
-                'schedule'=>ScheduleData::where('id', $split[2])->get(),
-                'choice'=>$choice,
-                'milestones'=>Milestone::where('id', $_REQUEST['milestones3'])->get(),
-                'pref_anest'=>Anesthesiologist::where('id', $_REQUEST['pref_anest3'])->get(),
-                'objectives'=>$_REQUEST['objectives3']
-            );
-        } else {
-            $input[2] = NULL;
+        // get current preferences
+        $currentChoices = array();
+        for($i = 0; $i < 3; $i++) {
+            if($split[$i] != 0) {
+                $schedule = ScheduleData::where('id', $split[$i])->get();
+                $currentChoices[$i] = array(
+                    'schedule' => $schedule,
+                    'case_procedure' => self::parseCaseProcedure($schedule[0]['case_procedure']),
+                    'milestone' => Milestone::where('id', $_REQUEST["milestones".($i+1)])->get(),
+                    'objective' => $_REQUEST["objectives" . ($i+1)],
+                    'anesthesiologist_pref' => Anesthesiologist::where('id', $_REQUEST["pref_anest".($i+1)])->get()
```

Comment:
Since `$_REQUEST` is used all throughout ScheduleDataController and REMODEL we might want to have a separate task to update all the occurrences to maintain consistency

Reply:
That's one approach. The other would be that since we're refactoring this file right now, we should make that change.

I'm fine with either so long as it happens.